langchain/src/text_splitter.ts

```diff
@@ -96,23 +96,39 @@ export abstract class TextSplitter
         let pageContent = chunkHeader;
 
         // we need to count the \n that are in the text before getting removed by the splitting
-        let numberOfIntermediateNewLines = 0;
+        const indexChunk = text.indexOf(chunk);
         if (prevChunk) {
-          const indexChunk = text.indexOf(chunk);
           const indexEndPrevChunk =
             text.indexOf(prevChunk) + (await this.lengthFunction(prevChunk));
-          const removedNewlinesFromSplittingText = text.slice(
-            indexEndPrevChunk,
-            indexChunk
-          );
-          numberOfIntermediateNewLines = (
-            removedNewlinesFromSplittingText.match(/\n/g) || []
-          ).length;
+          if (indexEndPrevChunk < indexChunk) {
+            const removedNewlinesFromSplittingText = text.slice(
+              indexEndPrevChunk,
+              indexChunk
+            );
+            const numberOfIntermediateNewLines = (
+              removedNewlinesFromSplittingText.match(/\n/g) || []
+            ).length;
+            lineCounterIndex += numberOfIntermediateNewLines;
+          } else if (this.chunkOverlap > 0 && indexEndPrevChunk > indexChunk) {
```

Comment:
Would there ever be a case where `indexEndPrevChunk > indexChunk` and `this.chunkOverlap` was zero?